dao/book_dao.py

```diff
@@ -18,11 +18,19 @@ def save(self, book: Book) -> None:
             raise Exception('O argumento passado não é do tipo Livro')
 
     @is_not_null
-    def find_by_title(self, title: str) -> List[Book]:
+    def find_all_books_with_title(self, title: str) -> List[Book]:
         if len(title) >= 2:
             books_with_title = list(
                 filter(lambda books: title.lower() in books.title.lower(),  self.list_book))
             return books_with_title
         else:
             raise('É necessário ao menos 2 caracteres para fazer a busca')
 
+    
+    @is_not_null
+    def find_by_title(self, title: str) -> Book:
+        book = next((book for book in self.list_book if title == book.title), None)
+        if book:
+            return book
+        raise Exception('Este título não consta no acervo')
```

Comment:
Não é engraçado como duas buscas, com resultados um pouco diferente, tenham implementações tão diferentes. Em um você precisa retornar um livro, na outra, uma lista de livros. 

- Em um, você faz o toLower, no outro não. 
- Em um você faz o len(), no outro não. (sim, eu sei que tem a regra de negócio que é diferente). 
- Em um você usa filter, no outro você usa o iterador next. 

É impressão minha ou você pode pegar o que é comum dos dois métodos e escrever um terceiro?

Reply:
> Em um, você faz o toLower, no outro não.
Erro meu, era pra adicionar. 

> Em um você faz o len(), no outro não. (sim, eu sei que tem a regra de negócio que é diferente).

O len existe na primeira implementação por causa da regra de negócio

> Em um você usa filter, no outro você usa o iterador next.
O filter também existe na primeira por causa da regra, já que retorna uma lista filtrada através do caractere passado. 

> Porque o uso do next?
Porque na segunda implementação estou utilizando um for com um generator, não uma lista, então a ideia é que ele vai retornar algo sobre demanda, por isso o uso do next. Achou? Ok, tudo bem, me retorna esse valor então. Caso não ache, me retorne None.

Reply:
Mas posso fazer algo mais trivial, que seria uma sintaxe de for normal. O generator é interessante só por causa do lazy evaluation, Se fossem muitos e muitos livros, a utilidade dele poderia ser mais evidente ao meu ver.

Reply:
- Eu saquei a ideia do iterator com lazy evaluation, mas por que você usaria aqui? Existe alguma necessidade para esse tipo de estratégia ou é só por que é legal? 😅 (mals, encho o saco mesmo)

- Você jogou um bom argumento: **se fossem muitos livros**. Pensando assim, por que não usar lazy evaluation em tudo então?

Reply:
Não foi porque é legal não kkkkkkkkk

Tudo que posso fazer com generator, eu posso fazer com lista, o que muda é só o fato do generator ser lazy evaluation e isso é muito útil para conjuntos de dados grandes, pode ser que foi prematuro utilizar um por agora. Já fiz um novo commit fazendo normal.